src/aolayer.cpp

```diff
@@ -120,9 +122,12 @@ void BackgroundLayer::load_image(QString p_filename)
 {
   play_once = false;
   cull_image = false;
-  if (ao_app->read_design_ini(
-          "scaling", ao_app->get_background_path("design.ini")) == "smooth")
-    transform_mode = Qt::SmoothTransformation;
+  QString design_path = ao_app->get_background_path("design.ini");
+  transform_mode = (ao_app->read_design_ini("scaling", design_path) == "smooth"
+                        ? Qt::SmoothTransformation
+                        : Qt::FastTransformation);
```

Comment:
You may find it useful to extract this into a function, since you make this conversion so frequently.

Reply:
see https://github.com/AttorneyOnline/AO2-Client/pull/302/commits/7f55932dbcb04fd5592449fec1b3d0a9c009e21e